modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4TcpChannel.java

```diff
@@ -34,6 +37,9 @@ public class Netty4TcpChannel implements TcpChannel {
     private final ListenableFuture<Void> closeContext = new ListenableFuture<>();
     private final ChannelStats stats = new ChannelStats();
     private final boolean rstOnClose;
+    private final Runnable flushTask;
+    // queue of objects that will be written by the next execution of #flushTask and the promise to resolve on their completion
+    private final Queue<Tuple<Object, ChannelPromise>> sendQueue = PlatformDependent.newFixedMpscUnpaddedQueue(128);
```

Comment:
Same queue as used by Netty itself. We don't really have any other good options available for single consumer anyway :)